pkgs/servers/plex/default.nix

```diff
@@ -1,84 +1,102 @@
-{ config, stdenv, fetchurl, rpmextract, glibc
-, dataDir ? "/var/lib/plex" # Plex's data directory must be baked into the package due to symlinks.
-, enablePlexPass ? config.plex.enablePlexPass or false
+# The actual Plex package that we run is a FHS userenv of the "raw" package.
+{ stdenv
+, buildFHSUserEnv
+, writeScript
+, plexRaw
+
+# Old argument for overriding the Plex data directory; isn't necessary for this
+# version of Plex to function, but still around for backwards-compatibility.
+, dataDir ? "/var/lib/plex"
 }:
 
-let
-  plexPass = throw "Plex pass has been removed at upstream's request; please unset nixpkgs.config.plex.pass";
-  plexpkg = if enablePlexPass then plexPass else {
-    version = "1.14.1.5488";
-    vsnHash = "cc260c476";
-    sha256 = "8ee806f35ccedcecd0cab028bbe1f7e2ac7de24292b715978d3165c4712f5c40";
-  };
+buildFHSUserEnv rec {
+  name = "plexmediaserver";
+  inherit (plexRaw) meta;
 
-in stdenv.mkDerivation rec {
-  name = "plex-${version}";
-  version = plexpkg.version;
-  vsnHash = plexpkg.vsnHash;
-  sha256 = plexpkg.sha256;
+  # This script is run when we start our Plex binary
+  runScript = writeScript "plex-run-script" ''
+    #!${stdenv.shell}
 
-  src = fetchurl {
-    url = "https://downloads.plex.tv/plex-media-server/${version}-${vsnHash}/plexmediaserver-${version}-${vsnHash}.x86_64.rpm";
-    inherit sha256;
-  };
+    set -eux
```

Comment:
You can probably drop the `set -x` here

Reply:
Good call!  Also, I took the opportunity to clean up some comments and quoting.